Runtime/Native/Windows/NativeClient.cs

```diff
@@ -0,0 +1,438 @@
+                                    AddNativeAttribute("error.type", "Hang");
+
+                                    NativeReport("ANRException: Blocked thread detected.", true);
+                                    // update error.type attribute in case when crash happen 
+                                    AddNativeAttribute("error.type", "Crash");
+                                }
+                            }
+                        }
+                        else
+                        {
+                            reported = false;
+                        }
+
+                        lastUpdatedCache = _lastUpdateTime;
+                    }
+                    else if (lastUpdatedCache != 0)
+                    {
+                        // make sure when ANR happened just after going to foreground
+                        // we won't false positive ANR report
+                        lastUpdatedCache = 0;
+                    }
+                    Thread.Sleep(5000);
+                }
+            });
+            _anrThread.IsBackground = true;
+            _anrThread.Start();
+            return;
+        }
+
+        public bool OnOOM()
+        {
+            return false;
+        }
+
+        public void PauseAnrThread(bool state)
+        {
+            _preventAnr = state;
+        }
+
+        public void SetAttribute(string key, string value)
+        {
+            if (string.IsNullOrEmpty(key))
+            {
+                return;
+            }
+            // avoid null reference in crashpad source code
+            if (value == null)
+            {
+                value = string.Empty;
+            }
+            AddAttributes(key, value);
+        }
+
+        public void UpdateClientTime(float time)
+        {
+            _lastUpdateTime = time;
+        }
+
+        /// <summary>
+        /// Read directory structure in the native crash directory and send new crashes to Backtrace
+        /// </summary>
+        public static IEnumerator SendUnhandledGameCrashesOnGameStartup(IEnumerable<string> attachments, IBacktraceApi backtraceApi)
+        {
+            // Path to the native crash directory
+            string nativeCrashesDir = Path.Combine(
+                    Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
+                    string.Format("Temp/{0}/{1}/crashes", Application.companyName, Application.productName));
+
+            if (string.IsNullOrEmpty(nativeCrashesDir) || !Directory.Exists(nativeCrashesDir))
+            {
+                yield break;
+            }
+            var crashDirs = Directory.GetDirectories(nativeCrashesDir);
+            foreach (var crashDir in crashDirs)
+            {
+
+                var crashDirFullPath = Path.Combine(nativeCrashesDir, crashDir);
+                var crashFiles = Directory.GetFiles(crashDirFullPath);
+
+                var alreadyUploaded = crashFiles.Any(n => n.EndsWith("backtrace.json"));
```

Comment:
Does each minidump get stored in its own directory?

Reply:
yes it is.